rx/subject/subject.py

```diff
@@ -30,23 +28,40 @@ def check_disposed(self) -> None:
         if self.is_disposed:
             raise DisposedException()
 
-    def _subscribe_core(self, observer: Observer, scheduler: Optional[Scheduler] = None) -> typing.Disposable:
+    def _subscribe_core(self,
+                        observer: typing.Observer,
+                        scheduler: Optional[typing.Scheduler] = None
+                        ) -> typing.Disposable:
         with self.lock:
             self.check_disposed()
             if not self.is_stopped:
                 self.observers.append(observer)
                 return InnerSubscription(self, observer)
 
-            if self.exception:
+            if self.exception is not None:
                 observer.on_error(self.exception)
-                return Disposable()
-
-            observer.on_completed()
+            else:
+                observer.on_completed()
             return Disposable()
 
+    def on_next(self, value: Any) -> None:
```

Comment:
If you inherit from `core.Observer` then you should not impl. `on_next`, but `_on_next_core` since `core.Observer` will then take care fo the stopped logic. Not sure I really like that base-class calling super-class thing, but it was again something inherited from .NET. We should at least keep it similar for all classes inheriting from `core.Observer` for now. Will it work?

Reply:
I added a commit which I hope addresses your comment (and I've rebased on latest master).

It looks like this works -- the tests all pass, anyway. But I take your point about calling super-class methods in this case, it does get slightly intricate: The `check_dispose()` calls should happen before looking at `is_stopped`, so the `Subject` class now overrides both `on_next` (to check disposed status) as well as `_on_next_core` (with the actual logic). Likewise, of course, for `on_error` and `on_completed`.

On the other hand, due to the same inheritance one level down, the `XxxSubject` subclasses don't need to know anything about this and need only override the `on_..._core` methods.